### ISSUE TITLE
testing on s390, ppcle Cannot run program "Xvfb"

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#4250]
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/157
```
06:45:20 dist:
06:45:21      [copy] Copying 1781 files to /home/jenkins/jenkins-agent/workspace/Test-Grinder/jvmtest/systemtest
06:45:21      [copy] Copied 712 empty directories to 8 empty directories under /home/jenkins/jenkins-agent/workspace/Test-Grinder/jvmtest/systemtest
06:45:21      [copy] Copying 10 files to /home/jenkins/jenkins-agent/workspace/Test-Grinder/jvmtest/systemtest/systemtest_prereqs
06:45:21 
06:45:21 build:
06:45:21 
06:45:21 BUILD SUCCESSFUL
06:45:21 Total time: 1 minute 14 seconds
[Pipeline] }
06:45:21 [ssh-agent] Stopped.
[Pipeline] // sshagent
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
running test in default mode
[Pipeline] stage
[Pipeline] { (Test)
[Pipeline] timestamps
[Pipeline] {
[Pipeline] echo
06:45:21 Running tests...
[Pipeline] wrap
[Pipeline] // wrap
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // stage
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
java.io.IOException: error=2, No such file or directory
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to ub16-390-1
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1741)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:357)
		at hudson.remoting.Channel.call(Channel.java:955)
		at hudson.Launcher$RemoteLauncher.launch(Launcher.java:1059)
		at hudson.Launcher$ProcStarter.start(Launcher.java:455)
		at org.jenkinsci.plugins.xvfb.Xvfb.launchXvfb(Xvfb.java:587)
		at org.jenkinsci.plugins.xvfb.Xvfb.setUp(Xvfb.java:697)
		at org.jenkinsci.plugins.workflow.steps.CoreWrapperStep$Execution.start(CoreWrapperStep.java:80)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:270)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:178)
		at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:122)
		at sun.reflect.GeneratedMethodAccessor293635.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
		at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
		at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
		at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
		at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
		at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
		at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
		at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:157)
		at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:155)
		at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:159)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
		at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
		at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
		at sun.reflect.GeneratedMethodAccessor99.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
		at com.cloudbees.groovy.cps.impl.ClosureBlock.eval(ClosureBlock.java:46)
		at com.cloudbees.groovy.cps.Next.step(Next.java:83)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
		at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
		at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
		at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$101(SandboxContinuable.java:34)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.lambda$run0$0(SandboxContinuable.java:59)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:58)
		at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:182)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:332)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$200(CpsThreadGroup.java:83)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:244)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:232)
		at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
		at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
		at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Caused: java.io.IOException: Cannot run program "Xvfb": error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at hudson.Proc$LocalProc.<init>(Proc.java:249)
	at hudson.Proc$LocalProc.<init>(Proc.java:218)
	at hudson.Launcher$LocalLauncher.launch(Launcher.java:936)
	at hudson.Launcher$ProcStarter.start(Launcher.java:455)
	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1313)
	at hudson.Launcher$RemoteLaunchCallable.call(Launcher.java:1266)
	at hudson.remoting.UserRequest.perform(UserRequest.java:212)
	at hudson.remoting.UserRequest.perform(UserRequest.java:54)
	at hudson.remoting.Request$2.run(Request.java:369)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE
```

**Comment by pshipton** [comment]
@jdekonin @AdamBrousseau @smlambert

**Comment by pshipton** [comment]
Occurred on ub16-390-2 as well. 
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/160/

**Comment by pshipton** [comment]
Seems the overnight builds are also impacted
https://ci.eclipse.org/openj9/job/Test-sanity.system-JDK11-linux_390-64_cmprssptrs/131
https://ci.eclipse.org/openj9/job/Test-extended.system-JDK11-linux_390-64_cmprssptrs/131/

and the functional tests
https://ci.eclipse.org/openj9/job/Test-extended.functional-JDK11-linux_390-64_cmprssptrs/132/

Also other platforms
https://ci.eclipse.org/openj9/job/Test-sanity.system-JDK11-linux_ppc-64_cmprssptrs_le/136

**Comment by AdamBrousseau** [comment]
[Looking at the playbooks](https://github.com/AdoptOpenJDK/openjdk-infrastructure/search?q=xvfb&unscoped_q=xvfb), I only see xvfb in the ubuntu yml. So I would say that infra was not ready for this change. That being said, our 390 machines at OpenJ9 are ubuntu so they _should_ have had it but I think the machines were created before it was added to the playbook and it was likely never added to the machines because it was originally only for jck testing.

**Comment by smlambert** [comment]
Its been in the playbook in some form since at least July 3, 2017 based on this [comment](https://github.com/AdoptOpenJDK/openjdk-tests/issues/49#issuecomment-312652929), but we will add additional restrictions on which tests use it, to avoid issues at openj9 where machine config is not same as Adopt (runs fine on 390 machines there).

**Comment by pshipton** [comment]
Should we just update the OpenJ9 machines to have it?

**Comment by smlambert** [comment]
Yes, I think we should update the OpenJ9 machines with the latest ansible scripts so that the machines across the projects are configured the same (to avoid strange differences in behaviour and slow downs in triage tasks).

Also think a regular practice of updating machines with latest ansible scripts should be instituted at both Adopt and OpenJ9 (and internally), on a transparent / published schedule for fire prevention, versus a firefighting approach to infra.

**Comment by pshipton** [comment]
@jdekonin what is the outlook for updating the machines?

FYI this also blocks omr acceptance builds.

**Comment by smlambert** [comment]
Added restrictions on which tests require display, should unblock regular (non-grinder) openj9 jobs which only run system functional: https://github.com/AdoptOpenJDK/openjdk-tests/pull/797  (eventually we will need to expand to other test groups / system and functional).

**Comment by AdamBrousseau** [comment]
See #4221.
We don't run the playbooks on the existing OpenJ9 ci machines at the moment. So even though [this change](https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/217) was merged last feb, it wouldn't have been installed on our 390 machines as they were setup before then.

I can go ahead and install/verify on the 4 - 390 & 4 ppcle machines via #4258
